ddtrace/internal/datastreams/kombu.py

```diff
@@ -0,0 +1,43 @@
+from ddtrace import config
+from ddtrace.contrib.kombu.utils import HEADER_POS
+from ddtrace.contrib.kombu.utils import PUBLISH_BODY_IDX
+from ddtrace.contrib.kombu.utils import get_exchange_from_args
+from ddtrace.internal import core
+from ddtrace.internal.datastreams.processor import PROPAGATION_KEY
+from ddtrace.internal.datastreams.utils import _calculate_byte_size
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def handle_kombu_produce(args, kwargs, span):
+    from . import data_streams_processor as processor
+
+    dsm_identifier = get_exchange_from_args(args)
+    payload_size = 0
+    payload_size += _calculate_byte_size(args[HEADER_POS])
+    payload_size += _calculate_byte_size(args[PUBLISH_BODY_IDX])
+
+    pathway = processor().set_checkpoint(
+        ["direction:out", "exchange:" + dsm_identifier, "type:rabbitmq"], payload_size=payload_size, span=span
```

Comment:
We should also have a `has_routing_key:{routing_key_present?}` edge tag to match java, see [here](https://github.com/search?q=repo%3ADataDog%2Fdd-trace-java%20type%3Arabbitmq&type=code#:~:text=173-,edgeTags%20%3D%3D%20%5B%22direction%3Aout%22%2C%20%22exchange%3A,%2C%20%22topic%3A%22%20%2B%20queueName%2C%20%22type%3Arabbitmq%22%5D,-227)